Add jenkins-slave-ansible-stacks

jkupferer opened the pull request:
#### What is this PR About?

This pull request adds a jenkins slave image for using ansible-stacks from a jenkins slave. This came out of work with a recent engagement where we provisioned projects and resources in OpenShift.

#### How do we test this?

See the README.md for instantiating the template to build the image. For example:

    oc process -f templates/jenkins-slave-ansible-stacks-template.json \
    -v SOURCE_REPOSITORY_URL=https://github.com/jkupferer/containers-quickstarts.git \
    -v SOURCE_REPOSITORY_REF=master \
    -v ANSIBLE_YUM_REPO=rhel-7-server-ose-3.3-rpms | oc create -f -

Next run the image in Jenkins with sample code to test. The jenkins service account will need self-provisioner role for the given example to succeed.

sabre1041 commented:
@jkupferer attempted to run through the PR and ran into a few issues:

1. Example pipeline needs to include a closing `}`
2. No playbook called `local-file.yaml` exists
3. It seems this PR is dependent on several PR's from being accepted/merged at [ansible-stacks](https://github.com/rht-labs/ansible-stacks/pulls)

Also, do we want to set default environment variables? Defining them within a BuildConfig as is done currently should overwrite the default value?

jkupferer commented:
Fixed up the missing closing curly.
Removed need for local-file.yaml.
Remove dependency in example in README for ansible-stacks PR for no-login feature.
Added default values to environment variables in the Dockerfile so that it can be simply built and tested outside of OpenShift.

sabre1041 commented:
lgtm
